src/sdl2/video.rs

```diff
@@ -1177,23 +1177,13 @@ impl Window {
         }
     }
 
-    pub fn surface<'a>(&'a self, _e: &'a EventPump) -> Result<&'a SurfaceRef, String> {
+    pub fn surface<'a>(&'a mut self, _e: &'a EventPump) -> Result<Surface<'a>, String> {
```

Comment:
What you can do here is create a WindowSurface that holds a SDL_Surface as well (but doesn't drop it in the end) and a mutable reference over Window. You can make it have exactly the same methods as surface, but with a `update_window(&mut self)` method added, which would effectively do [this](https://wiki.libsdl.org/SDL_UpdateWindowSurface).

This is just a suggestion and I'm just thinking it would be nice to have that, don't feel obligated to implement it word by word or implement it at all, you can adapt this to your liking. I'm thinking this because `update_surface` is currently very awkward to use and isn't very explicit ... See [here](https://github.com/nrxus/rust-sdl2/blob/09153e52cfe94481bc95d8af7e1b15617e890c7f/src/sdl2/video.rs#L1190-L1208). It would be better to implement these in the given Surface instead of in the window.